controllers/fenceagentsremediation_controller.go

```diff
@@ -120,7 +120,7 @@ func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ct
 
 	// Validate FAR CR name to match a nodeName from the cluster
 	r.Log.Info("Check FAR CR's name")
-	node, err := utils.GetNodeWithName(r.Client, req.Name)
+	node, err := utils.GetNodeWithName(r.Client, getNodeName(far))
```

Comment:
I can't add the comment in the exact line, so writing here instead.
At line 129 the error log message should be updated to

from  "Didn't find a node matching the CR's name" (CR name is not expected to match node name anymore)
to something like "Could not find CR's target node"